### PR TITLE
LPD-28210 Set the PrincipalThreadLocal in ObjectEntryInfoItemFieldValuesProviderTest

### PR DESCRIPTION
**Pull request by vngoliveira**
[LPD-28210](https://liferay.atlassian.net/browse/LPD-28210)

### CODE REVIEW

**Comment by vngoliveira** [comment]
ci:test:object

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFieldValuesProviderTest.java
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFieldValuesProviderTest.java:1]: @vngoliveira did you try to add the `PermissionCheckerMethodTestRule` instead (like [here](https://github.com/liferay/liferay-portal/blob/eea2628408e5340d5e4e30e112ed24e443b9a103/modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/graphql/v1_0/test/ObjectDefinitionGraphQLTest.java#L74))?

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFieldValuesProviderTest.java [hunk 1]
```diff
@@ -173,6 +175,15 @@ public void setUp() throws Exception {
...
 			ObjectRelationshipConstants.TYPE_ONE_TO_MANY, null);
+
+		_originalName = PrincipalThreadLocal.getName();
+
+		PrincipalThreadLocal.setName(TestPropsValues.getUserId());
+	}
+
+	@After
+	public void tearDown() {
+		PrincipalThreadLocal.setName(_originalName);
```
Comment (carolmariaabb) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/info/item/provider/test/ObjectEntryInfoItemFieldValuesProviderTest.java:186]: @vngoliveira if the `PermissionCheckerMethodTestRule` does not work, please change the code to something similar to [this other test](https://github.com/liferay/liferay-portal/blob/eea2628408e5340d5e4e30e112ed24e443b9a103/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipServiceTest.java#L97-L107), i.e.,

- Also save the `_originalPermissionChecker`;
- We usually call the `PrincipalThreadLocal.setName` and the `PermissionThreadLocal.setPermissionChecker` inside the test method, and not inside the `setUp` method;

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**